### ISSUE TITLE
List of games that don't work 100% (using an iPod Touch)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ifrotz/iosfrotz#203]
```
I am currently using Frotz 1.6 (the latest at the time of writing) using an 
iPod Touch iOS 5 (or whatever Apple's calling it. Also, don't ask me about the 
gen. It was bought in January 2013 and let's leave it at that).

This is to be a list of every game that doesn't, at present, really work 
smashingly on Frotz, as well as the reason why, as far as I could tell. I will 
add to it as I play the games. This list will include known problems, such as 
the .z6 games, because it's as well to have all these things in a big list and 
watch it dwindle to nothingness. :)

Let's begin.

007 - Alto Secreto: Crashes, simply put.
222 Contra el Cangrejo: after some seconds of working nicely, some weirdness 
ensues. The line that indicates where my text will go jumps up and down, I stop 
seeing what I'm writing, and before long, crash.

Incidently, there may be some issues with some Spanish games, probably the same 
issues that there are in Git (nevertheless they work fine on Glulx).

Balances (Glulx version): the problem is mostly due to the size of the screen, 
and the size of the hyperlinks. If we could pinch/zoom them, as per another 
suggestion, it would be perfect.

Glkchess: the display is quite off. Images are chopped in the middle. Without 
pinching, it's also not really playable. With pinching, however...

zRogue: I know, I know, it's an abuse, you don't *have* to run it. But it 
starts to display surprisingly well. Then something goes FUBAR and it 
completely stops responding. At which point there's nothing to do but to 
totally remove the game with DiskAid; because Frotz will save it as is, in its 
unresponsive state. ;)

Z-Chess: completely unplayable without arrow keys. I understand this is very 
specific, so a suggestion would be to define, in Frotz's settings whether not 
the arrow keys were currently emulated, and let the user define which keys 
would do, for the duration of the session. It's a bit of a hoop, yes, but very 
few games require this (which I know is an argument against it being 
implemented at all, naturally).

I will keep you posted as I play (might be a while since I'm currently playing 
a puzzle fest!). Feel free to add to this.
```

Original issue reported on code.google.com by `ilprimoc...@gmail.com` on 27 Jan 2013 at 11:16

**Comment by PeterPiers** [comment]
Narcolepsy (http://ifdb.tads.org/viewgame?id=b67iuep7o6b0kxpm): I'm not sure what you can do about this one, but here goes...

It's designed to have an overlay that is accessible through an external .cfg file, and the lines:

WindowFrame=no
WindowMask=3

...so I'm not sure how you'd work around this. Without that, the game is playable, but not as pretty as it was meant to be.

There are two text input windows. Well, there's one text window (the main one) and a secondary one, and input switches between the two. The game starts in the secondary one. When input is expected for the secondary window, the text doesn't appear as you're typing - the command line for the secondary window remains blank. But when you press enter your command does run. No autocomplete for this one, naturally.

Also, the secondary window doesn't scroll automatically. If you want to scroll it, even after a command, you have to do it manually. Sometimes this is ok, sometimes it isn't. Hard to automate, so maybe it'd be best to keep it as it is.

(also, it's a shame about all the "deleted comment" entries!)

**Comment by ifrotz** [comment]
I've deleted the extraneous [deleted comment] entries, which were introduced by the migration process.
You might be able to change your workflow here and just edit the comments instead of deleting and reposting.

**Comment by PeterPiers** [comment]
Oh yes, that'll be much better. :) Cheers!

**Comment by PeterPiers** [comment]
Nazy Jones (http://www.leavventure.altervista.org/nazy.html, http://www.avvtestuali.altervista.org/scarica/NazyJonesEIDittatoriDellArcaPerduta.zip): Same formatting issues as seen above, in fixed-length width. 

![img_0486](https://cloud.githubusercontent.com/assets/12926960/8197515/d5cfeb7e-1490-11e5-9882-3905e5ec4fe4.PNG)

**Comment by PeterPiers** [comment]
Nightfall (http://ifdb.tads.org/viewgame?id=3a4kqs374jqu2li6): Same formatting issue as above. I can provide the screenie, but it's pretty much identical.

Also, a small other detail. I don't think you can do anything about it, but it's reproducible and seems to be a matter of interpreter speed/optimization, so I'll go ahead and include it here. Except that I can't include savegames in this place, it seems... 

![go north](https://cloud.githubusercontent.com/assets/12926960/8266751/9739d722-1737-11e5-97dd-b74665b9164f.jpg)

**Comment by PeterPiers** [comment]
And just to stop the flow of nitpicking - I've just realised that iFrotz is significantly faster than zarf's own iOS framework. There's a bit of a delay between your command input and the game's response - it's small, but it's very noticeable because in iFrotz it's instantaneous.

So, kudos! I may have griped about a possible speed optimisation in the comment above, but make no mistake, as it is iFrotz is already superb on that aspect. Zarf's own terp isn't as fast, and that should tell you something!

**Comment by PeterPiers** [comment]
Noir d'Encre (http://ifdb.tads.org/viewgame?id=ylkl8140si1wxttr): Plays sound. Which, incidently, seems highly praised in a review.

**Comment by PeterPiers** [comment]
Notte prima dell'esame (http://ifdb.tads.org/viewgame?id=d5bj9rohyxx2w7r): It says it plays sound.

**Comment by PeterPiers** [comment]
Obituario (http://ifdb.tads.org/viewgame?id=3hwdbja9hj4kv49u), "Deluxe" edition (i.e., Glulx): Plays sound.

**Comment by PeterPiers** [comment]
Ocaso Mortal (http://ifdb.tads.org/viewgame?id=e4sfma9el697c96h): Formatting issues. The sort where, in fixed-width font text, the printed line just goes on and on offscreen. I don't think you'll need a screenie, but it's available should you want one.

**Comment by PeterPiers** [comment]
Old Fogey (http://ifdb.tads.org/viewgame?id=w00uocj8hnmdcuzf): Plays sound.

**Comment by PeterPiers** [comment]
Olivia's Orphanarium (http://ifdb.tads.org/viewgame?id=waljztz5tm1at7x6): Formatting issues. 

I'm attaching screenies of this one because the previous ones were of the "zoom out and it'll be fixed" sort, whereas here a funny thing happened. I zoomed out and it was fixed; I zoomed out some more and it was incorrectly formatted again. So it's not like the problem is exclusively having the font size too big.
![img_0503](https://cloud.githubusercontent.com/assets/12926960/8408139/5fb2fa24-1e65-11e5-9c6e-00d55686136c.PNG)
![img_0505](https://cloud.githubusercontent.com/assets/12926960/8408140/638123e2-1e65-11e5-980a-2a3022912a4a.PNG)
![img_0504](https://cloud.githubusercontent.com/assets/12926960/8408141/64dc9b4a-1e65-11e5-8222-93f187a09ab9.PNG)

**Comment by PeterPiers** [comment]
Once upon a dream (http://newsletter.aifcommunity.org/2010_AIF_Mini-comp.zip) (AIF): Plays sound

**Comment by PeterPiers** [comment]
One Eye Open (http://ifdb.tads.org/viewgame?id=qgdgtaroybcdnmi): A curious formatting issue where text overlaps. It's been a LONG time since last I saw this one. The interesting this is, it's totally reproducible! The screenie shows the problem in action; if I zoom out the text no longer overlaps, if I zoom back in it does. 

When I say "reproducible", I don't mean that every time that I try the command in-game it renders the bug. The screenie clearly shows that doesn't happen. But every time I restart the game and enter "HINT" as my first command, being that my font is Baskerville and the font size is upped to the max (currently 24)... it will ALWAYS happen. I'd seen something like this before, but it had never, ever been reproducible.

![img_0508](https://cloud.githubusercontent.com/assets/12926960/8420987/edbdd4f0-1ec0-11e5-9a73-1d39a08e53d4.PNG)

**Comment by PeterPiers** [comment]
Oppositely Opal (http://ifdb.tads.org/viewgame?id=z3scyumvgsr75blm): Like Jack Toresal, for some reason, autocomplete doesn't work in this game.

**Comment by PeterPiers** [comment]
Order (http://ifdb.tads.org/viewgame?id=6kpk3744vsj7wu1v): Formatting issues of the sort where the line just runs off the screen. Not literally. It would be cool if it were, though. Literally, I mean.

This is seen in pretty much the entire HINT system (menu-based), which is all in fixed-witdh.

**Comment by PeterPiers** [comment]
Oxygen (http://ifdb.tads.org/viewgame?id=jz2q7uk4gp0zx2ph): Plays sound

**Comment by PeterPiers** [comment]
The Mean Story (http://ifdb.tads.org/viewgame?id=aixamxezzos7b4e6): Freezes on loading.

I think I know why, and it's the same as reported above: the room name is too big. It looks like iFrotz doesn't know how to deal with very long room names, and it just freezes completely. The room name in this one is "The Blackest Of Black Markets In The Third (Or Indeed Any Other) World". Compare above with Mortlake Manor, where it froze up upon entering a room called "Intersection of a 
North-South Corridor and an East-West Passage".

**Comment by PeterPiers** [comment]
Mahadev (http://ifdb.tads.org/viewgame?id=v8fqg8rj3pn5nvz): Load up the following savegame.

![know](https://cloud.githubusercontent.com/assets/12926960/8768508/a8911ad0-2e7a-11e5-8378-02a3c4025455.jpg)

Yes, just as before, I had to upload it as an image file, but the trick seemed to work well. Just download as .sav, or rename after it's been downloaded.

Once it's downloaded, restore it and type "know". The issue is that the text in the quote box is too long and iFrotz doesn't display it properly, cutting some letters out.

Mind, it's a ridiculously huge quote. It shouldn't be there in the form that it is. Regardless, someone made it that way, and it does show that iFrotz has a limit on the length it can handle in quote boxes.

This happened even if I resized to the smallest font sizes, incidently.

**Comment by PeterPiers** [comment]
Paint and corners (http://ifdb.tads.org/viewgame?id=3smmikbeodg0tqzz): Requires arrow keys. Also runs a bit sluggish.

It's also an ambitious abuse of the ZMachine, so I don't see it being a priority. But it's cool that it works at all, and apart from the issues above, it displays very decently.

**Comment by PeterPiers** [comment]
Pánico a la Caída (http://www.caad.es/fichas/panico-la-caida.html): Crashes on loading. I really don't know why. I'm enclosing a copy of the beginning, in text form (the bit between [brackets] is a quotebox), in case you spot something that'll explain this. Surely it can't be the quote marks or the accented "i" in the title, can it?

***

[ Todavía no me puedo creer lo que he visto. Estoy seguro de que van a venir a por mi.
  Nunca debí asomarme a la ventana de esa casa y creerme lo que me decían de que habitaban sectarios en ella.
  Ahora voy a tener que cambiar de ciudad, de trabajo, de amigos, de vida.
  Posiblemente alguna persona de todas las que están aquí, en esta terminal, me está siguiendo.
  No me van a dejar que les delate...
  Lo mejor que puedo hacer es no llamar mucho la atención y permanecer sentado en la silla hasta que nos dejen entrar en el avión.]

"Pánico a la caída"
Realizado por: David Ruiz

(C) 2010 para la participación en la IV Minicomp organizada por CAAD (Aunque no llegó a presentarse a tiempo xD )
Revisión 1 / Número de serie 101224 / Inform v6.30 Librería 6/10E 

Terminal del aeropuerto (en la silla)
Estas sentado en una silla, en la segunda de las dos filas de sillas que hay. Desde aquí tienes una buena visión de la terminal. 
El hangar es bastante frio, una puerta grande, al norte, conduce a la pista de despegue, y justo al lado hay un mostrador con una azafata sonriente, al sur desandarías tus pasos para salir del aeropuerto y una pequeña puerta de madera permanece cerrada al este, a diez metros, justo enfrente de ti. 
Unas veinte personas se encuentran aquí contigo y no se te quita de la cabeza que posiblemente alguna de ellas venga a por ti...

>

**Comment by PeterPiers** [comment]
Patanoir (http://ifdb.tads.org/viewgame?id=si9s1jktywxj5vdk): Transcript doesn't work properly.

This is definitely strange. Surely it must affect some other games as well, but none that I've tested, none that I've found. Load up PataNoir and start a transcript (usually done on the first turn), and play a few turns, then check the transcript. The commands are off by one. The first command is repeated. So that if you type "x me, look, i", you get an output like:

"x me"
(response to x me)
"x me"
(response to look)
"look"
(responde to i)

Real strange, this one.

**Comment by PeterPiers** [comment]
Pathfinder (http://ifdb.tads.org/viewgame?id=fktbcvczqdgeh2al): Formatting issues. Both; the "words run off the screen in fixed-width" on and the "line breaking in the middle of the words for the rest of the paragraph". Screenies available if you want them.

**Comment by PeterPiers** [comment]
PDFA Ottumwa (http://ifdb.tads.org/viewgame?id=1cq6xseeyl1y1wsd, second link - the first Glulx one): Another one where autocomplete doesn't work. Bizarre.

**Comment by PeterPiers** [comment]
Peril in Pleasantville Part 2 (https://mega.co.nz/#F!7hdklb7I!V78elxci49ydRBO2LKgF5Q - link goes to my own collection, the original isn't around much anymore). Doesn't load; crashed iFrotz.

I guess it being 200mb bit may have something to do with that. ;) It's an AIF game, so you can guess what the bulk of THAT is. Graphical, of course.

But, content notwithstanding, it should run on iFrotz. If it IS because of the size, well, it's a warning signal - other games may be just as big with more tasteful graphical content.

Another thing: every time I opened the game file in FileApp, it imported a NEW copy of the game onto iFrotz. This *never* happens with other games; I try to import something that already exists, iFrotz usually - rightly! - opens the game instead of importing again. But in this case I ended up with iFrotz having three 200mb games, all with the same name (reduced my storage space from 600mb to 0mb, that's how I'm absolutely certain)

**Comment by PeterPiers** [comment]
Phantom of the Arcade (http://ifdb.tads.org/viewgame?id=7ylhba37cfbks70): Formatting issues, the sort where the line breaks get screwed up for the rest of the paragraph. Screenie if you want it. It only happened the once.

BTW, I considered putting these in a separate ISSUE. I figured that with this list you can "easily" (well, Ctrl-F easily) see how many games are afflicted with a certain problem, so you can prioritise it for the next release. But if you'd rather, since this one is so specific, I can create a new ISSUE for it, add examples to that one instead, and delete them from this one.

**Comment by PeterPiers** [comment]
Photopia (http://ifdb.tads.org/viewgame?id=ju778uv5xaswnlpl): The Glulx version doesn't change the colours properly. The ZMachine version does.

The game's age may have something to do with it - it's quite possible Glulx wasn't quite the same thing back then as it was today. The issue seems to be that whenever the background AND font colour need to be changed only of them is actually processed; then, AFTER the room description begins to print, it changes the other one.

This is hard to describe but easy to show. Load up the Glulx version. The first thing you'll see is that the background colour remains your chosen default - but it should have been WHITE instead. And indeed after you anser the first question (do you want instructions?) it does change to white.

Then you go "z.z" and trigger the first chance. You get a graphic shown. After the graphic, the background colour's changed to black. And your font is supposed to have changed to red, but instead it remains black. It only changes after the introductory text for this new sequence, just in time to print the first room description.

End this section by typing "n.e.s.s.w.get pod.n.u". Then you'll see that the background colour changes... to YOUR default; it should change to white. Type a command; it's still in RED. Press ENTER; your command changes to black and the background changes to white.

New section, "out.d.z.z" to finish. Same as the red section - the background changes colour but the font doesn't, not until it's time to display the room description making the introduction unreadable.

**Comment by PeterPiers** [comment]
Piracy 2.0: Issue with text not word-wrapping properly AND text running off the screen.

I'm going to stop listing every individual case of these two bugs. Firstly, because I've missed a few, some games I forgot to report. But mostly because I guess you've seen them enough, you know about them, and when you fix it for one game it'll be fixed for all. No point in bogging this down; best to leave this issue open for more important stuff.

**Comment by PeterPiers** [comment]
Playing Games (http://ifdb.tads.org/viewgame?id=7xxhw9da4j3ff3hd): Plays sound. It is entirely optional, and not at all crucial for solving the game - a mere convenience.

**Comment by PeterPiers** [comment]
Plundered Hearts Invisiclues (http://www.waitingforgo.com/invisiclues/plundizm.z5): It's only the InvisiClues, mind, which is a .z5 file made by a third party at www.waitingforgo.com.

Trying to open a hint crashes iFrotz, unless it's in landscape mode and the font is pretty darned small. I think it has to do with the sheer amount of text that ends up in the status bar, those questions are VERY verbose. Regardless, I'm sure iFrotz could deal with that more elegantly than crashing. ;)

**Comment by PeterPiers** [comment]
Puj (http://www.ifarchive.org/if-archive/games/competitions-spanish/xcomp2008/puj.zip): Plays sound. Probably not critical for finishing game.

**Comment by PeterPiers** [comment]
Pytho's Mask (http://ifdb.tads.org/viewgame?id=spsnevz1br6f2z3v): The ZCode version has a problem which is actually the same as the Plundered Hearts invisiclue above. The game tries to print a LOT of text into the status bar (type "z" a few times), and it just crashes.

I also found an issue in the Glulxe version, where it wasn't updating an extra window properly, and I could reliably replicate it... but then I tried the ZCode version, it crashed so ferociously I had to reset my iPos (Home+on/off button), and then when I quit and reloaded iFrotz I found myself unable to replicate it with the saved game I had. So... yay?

**Comment by PeterPiers** [comment]
Autumn Leaves (http://meadstelzer.com/daniel/if/sketchbook/leaves/): A card game (!), that makes use of some special characters iFrotz doesn't display.

You can try it online; it displays properly on Parchment and on my desktop terp. But in iFrotz (type DEAL to start) all the fancy characters are replaced by question marks (so you can't, say, see any card's suit).

Otherwise, it actually works pretty well.

**Comment by PeterPiers** [comment]
Bureaucrocy (http://ifarchive.org/if-archive/games/zcode/Bureaucrocy.zblorb), R4: "Fatal error: Text buffer overflow" on loading. Doesn't happen on my desktop terp.

**Comment by PeterPiers** [comment]
Beyond Division (http://ifdb.tads.org/viewgame?id=vk4j4zp5slsh9jyp): Mysteriously, another one where autocomplete doesn't work!

**Comment by PeterPiers** [comment]
Ghosterington Night (http://ifdb.tads.org/viewgame?id=6kko9va64c4o5jan): Newest release plays sound

**Comment by PeterPiers** [comment]
La valée mystérieuse: plays sound.

**Comment by PeterPiers** [comment]
Scroll Thief (http://ifdb.tads.org/viewgame?id=o6kvclutag67skou): I have not yet play-tested this one, but see the following exchange: http://www.intfiction.org/forum/viewtopic.php?f=19&t=18399&view=unread#p97725

I'm not sure what I'm reading, to be honest. An in-game loop? Stack overflow? iFrotz crashed and after restarting it was unstable?

From my experience, whenever iFrotz *does* crash, it's rarely 100% reliable when it's reloaded, especially when it tries to continue from where it crashed. I have best performance, in those cases, when I open another game - one that I know has no problems - then maybe I close iFrotz myself and reopen it and reopen the game I was playing.

I never mentioned this because I can't replicate it, and who knows what might be behind it and how game-specific it might be.

**Comment by PeterPiers** [comment]
¡Qué noche! (http://ifdb.tads.org/viewgame?id=m8zhthj0gj2mkc63): Plays sound

**Comment by PeterPiers** [comment]
Regreso al Eden (http://ifdb.tads.org/viewgame?id=r76xars2xn8jn8j7): Plays sound.

**Comment by PeterPiers** [comment]
Remakorp (http://ifdb.tads.org/viewgame?id=diziwpamsxpbkzxg): Same colour bug as Photopia (Glulx). Load up the game to see a green font over your selected background, and press a key a few times to go over the introduction. Then you get the cover image. Press ENTER and you'll see the same green font... over a green background.

What's happening is, after the last of the "green font" text is displayed - and that happens after the cover image - the display switches to a green background and white font. But it's like the switch is happening too early, and most of the screen gets prematurely covered in the green background... including the letters that were still drawn with a green font.

**Comment by PeterPiers** [comment]
Return to Zork: Another Story (http://ifdb.tads.org/viewgame?id=1b82dexzzu81ero0): Again, one location's name is too big and iFrotz chokes on it, even if I scale the whole text down to the minumum size before entering the room. The room's name is "[Flying Over the Desert (again and again) - hanging by a vulture]" (brackets included).

![03-flying](https://cloud.githubusercontent.com/assets/12926960/10997158/ae9c2628-8482-11e5-900f-4ed4c605fa53.png)

This is the saved game (rename to .sav), just type Z once or twice and you'll see the bug in action.

**Comment by PeterPiers** [comment]
Revenger (http://ifdb.tads.org/viewgame?id=mr67fiyrvgai97ms): Same bug: too much text in the status bar. This time it's the Hint system. Type "hints" (twice) to get to the hint system, select the first cathegory. Some of those options freeze iFrotz because it's too much text. Even if I reduce the size to the unreadable minimum.

**Comment by PeterPiers** [comment]
Robin & Orchid (http://ifdb.tads.org/viewgame?id=ahhlbqy5f39wd1x): No autocomplete.

**Comment by PeterPiers** [comment]
Andromeda 1983 (http://ifdb.tads.org/viewgame?id=k7gryitw24knktcc): Plays sound

**Comment by PeterPiers** [comment]
Peril in Pleasantville Part 2 - Doesn't load; crashed iFrotz.

It's available in the AIF newsletter website as a torrent: http://portal.aifcommunity.org/games/inform/Peril_in_Pleasantville_Part_2.torrent. That's because it's +200mb. I can also, alternatively, if you like, put it up temporarily in MEGA or GoogleDrive.

I thought that it had to do with the sheer size, but another AIF game (same author) also exceeds 200mb - is in fact about 15mb larger - and plays OK, so that's not it.

I also reported a strange issue last time with this game not being added to the playlist. Well, it's being added correctly now. Don't know what was up with that, but it's no longer an issue anymore.

**Comment by PeterPiers** [comment]
Coffee Date (http://www38.zippyshare.com/v/BOnPBakA/file.html): Plays sound. In fact, it seems to revolve entirely around a conversation which exists only in audio files, so it's unplayable without sound support.

Mind you, this will only be a real blow to anyone who was just dying to play a +200mb AIF game centered around a date and its "payoff"... still, it's an Inform game, and it requires sound to play through successfully.

**Comment by PeterPiers** [comment]
Heezy Park (http://ifdb.tads.org/viewgame?id=2izj2y1jraoyzqev): At the beginning, type >X MEGA and then answer >Y to the question you're asked. You're treated to some asterisks making some simple ASCII art.

When using the Baskerville font (my personal choice), it's completely misshapen, the way it usually happens when a variable-width font is used for ASCII art. Predictably, when I selected a fixed-width font, it displayed correctly.

So why am I reporting this? Because in Windows Frotz, which also uses my personal favourite Baskerville font, _the ASCII art displayed perfectly_. Perfectly spaced, except in the scrollback. So it looks like the game was actually setting the font to be fixed-space before drawing the art, as it should - but in iFrotz it still looks messed up.

**Comment by PeterPiers** [comment]
Hard Puzzle (http://ifdb.tads.org/viewgame?id=81afnluyffpxa8iv): This is reproducible, it happened every time. Type >ABOUT and check the screenshot provided. I'm using font size 24, on an iPod, Baskerville font.

![img_0583](https://cloud.githubusercontent.com/assets/12926960/12660802/9e849d5c-c60d-11e5-90db-608306b0ea42.PNG)

**Comment by PeterPiers** [comment]
Le Orme del Chisciano (http://www.oldgamesitalia.net/sites/default/files/avventuretestuali/Le_Orme_Del_Chisciano_Full.rar): Plays sound.

Also, the display is too cluttered for the game to be playable on an iPod, but I mention that only in passing; I'm sure it works great on an iPad, so it's not anything iFrotz can do anything about. Still tries to play sound and complains about it being unsupported, though.

**Comment by PeterPiers** [comment]
La Legión de las Tinieblas (http://www.caad.es/fichas/la-legion-de-las-tinieblas.html): Autocomplete doesn't work

**Comment by PeterPiers** [comment]
There's definitely something going on. When I spoke about Hard Puzzle, I mentioned a problem with "illegal object number"...

...and when I tried "Map" (http://ifdb.tads.org/viewgame?id=wttpef6vhl48hepx) I had a similar issue. Objects wouldn't appear. Test it by typing "e.s.x bed"; I would get "Which do you mean, or ?".

I restarted a few times, it was always like this. I tried it hours later on my desktop terp, it displayed correctly. I tried again on iFrotz, and it was working OK now.

Next time something like this happens I'll try saving the game to see if the savegames retain the error. Also, I believe Hard Puzzle and Map are both from the same author. Also, as you know I have hundreds (literally) of games in iFrotz, and I don't know whether that has anything to do with it.

I'll keep my eyes peeled. If I notice a pattern I'll open a bug report proper.

**Comment by PeterPiers** [comment]
Return to Pleasantville (http://www.filefactory.com/file/4l5n65jyfp1x/RTP.rar): Crashes on loading. Also, it will allow you to import it into iFrotz multiple times, creating copies every time, instead of just trying to open the copy it already has.

**Comment by PeterPiers** [comment]
SameGame (http://ifdb.tads.org/viewgame?id=ph0y9c8b9ugvxdrv): Works well, especially considering it's a ZMachine Abuse, but has two lacking features.

It's supposed to also allow for mouse-clicks. I've seen this not being allowed before, in .z6 games ("game", actually. Journey). I guess mouse support is not the same for the ZMachine as it is for Glulx?

And at the end it tries to write your highscore into an external file, and then complains that it couldn't do it. Again, I know this is possible in Glulx games (Alabaster and Kerkerkruip, for instance, work beautifully), so again maybe it's just the ZMachine?

Anyway, this has got to be the lowest priority of the whole lot. But hey. Being thorough.

**Comment by PeterPiers** [comment]
Saloon (http://www.caad.es/fichas/saloon-vr-2.html): Plays sound

**Comment by PeterPiers** [comment]
Sarvegne (http://ifdb.tads.org/viewgame?id=7gipckvj6bvv988g): Glulx version plays sound

**Comment by PeterPiers** [comment]
Schizo (http://ifdb.tads.org/viewgame?id=w4a5xqvu6hiy7vjg): Plays sound.

It also tries to be too fancy, with extra windows, so it's not really playable on an iPod, but I expect a larger device won't have any trouble.

**Comment by PeterPiers** [comment]
Salvate lo Stregatto (http://ifdb.tads.org/viewgame?id=1elq80q2fb91e9dd): For some odd reason, autocomplete doesn't work in the Glulx version of this game.

**Comment by PeterPiers** [comment]
Scopa (http://ifwiki.org/index.php/Scopa): A beautiful ZMachine abuse, replating a card game. Played through the mouse.

It doesn't actually load on my iPod Touch because the screen is too small, but there's two issues I can see and one I can guess at. They all have to do with the game being .z6.

I can see that the game doesn't get loaded into the playlist, like Moments Out Of Time and other .z6 games. I can open it using another app's "Open With" function, like FileApp, but then I can't select it from iFrotz's own playlist.

I can see that some of the graphics are in a separate .blb file. iFrotz can't deal with this, and it's a huge pain for .z6 support that it's done this way. I don't know of a way to bundle those graphics together with the story file in a way that an interpreter may read correctly...

...and I can guess that the mouse won't work. It's completely mouse-driven, and as I understand there's no mouse support for the ZMachine in iFrotz. Not many games use them, so meh.

**Comment by PeterPiers** [comment]
Sensory Jam (http://ifdb.tads.org/viewgame?id=0ulrnusrwihmwefi): Images don't resize properly to the image window, as reported in some other games before, so the image gets cut off. Also, plays sound.

This is just for completion - after all, SJ is a test suite of sorts, and Glulx has moved on since. It may no longer be at all relevant. Still. For completion.

(once we get sound support I'd be glad to consider this as done, since you've indicated you've done as much as you could towards graphics support and it's very likely this issue won't show up on larger devices - and if it does, you'll be able to see for yourself!)

**Comment by PeterPiers** [comment]
Sequitur (http://ifdb.tads.org/viewgame?id=hnta8r6vv47typj9): Plays some sound

**Comment by PeterPiers** [comment]
The Shadow in the Cathedral (http://ifdb.tads.org/viewgame?id=g79qfkq3m3dtffq4): Autocomplete doesn't work

**Comment by PeterPiers** [comment]
Sherlock: The Riddle of the Crown Jewels (http://ifdb.tads.org/viewgame?id=j8lmspy4iz73mx26): Plays sound. Infocom game, so sound is bundled separately, so this one may never really work. Sound isn't critical for this game, anyway; just adding this entry to be complete.

**Comment by PeterPiers** [comment]
The Black River Emerald (http://ifdb.tads.org/viewgame?id=05aqnleagcdwgh9w): Doesn't play. Ok, so it's a .z6 game and therefore unsupported, but it's still curious, because other .z6 games play (even if they don't show up on the games list - a bug already reported). In this one I actually get the message:

**Unreadable story file**
Frotz doesn't understand the format of this file
_Drat_

First time I've seen _that_.

**Comment by PeterPiers** [comment]
You can delete the one for "Mortlake Manor (http://ifdb.tads.org/viewgame?id=5acwwdfbclwq9bj4): Freezes." It's solved now.

**Comment by PeterPiers** [comment]
Pánico a la Caída (http://www.caad.es/fichas/panico-la-caida.html): Autocomplete doesn't work. But up until recently it wouldn't even load, so well done you!

**Comment by PeterPiers** [comment]
Shogun (Infocom's): Actually seems to play well! Though no graphics, and without mouse support or arrow keys the hints and other menus are hardly usable.

I didn't test it that much because the graphics aren't showing, and it's a known issue that it's just one of those very specific games unlikely to get much support. Just putting it here for completeness, and to say that, well, it does run!

**Comment by PeterPiers** [comment]
Signal Error (http://ifdb.tads.org/viewgame?id=32id49lmaviphkas): Work great, but as you play it becomes slower and slower and slower. It's always playable, but by the end of the game it starts to cross the linto into "too much". Savegames available.

The game does do a bunch of stuff - typo correction, pathfinding, it goes out of its way to provide a friendly experience. It may not be possible for you to do anything about this, unless iFrotz can be optimised in some way?... The slowdown is not present on a desktop.

**Comment by spathiwa** [comment]
If you save the game, then quit and restart and restore the save, does it continue to be slow?

**Comment by PeterPiers** [comment]
Yes, though not as grindingly slow as I seem to remember it from yesterday. But there's a noticeable difference between games saved near the beginning (not instant, but perfectly acceptable) and near the mid- to end-game (veering into frustration territory).

**Comment by PeterPiers** [comment]
Six (http://ifdb.tads.org/viewgame?id=kbd990q5fp7pythi): Plays sound

**Comment by PeterPiers** [comment]
Slenderman - L'incubo (http://creepyfiction.blogspot.pt/2014/07/slenderman-lincubo.html): Plays sound

**Comment by PeterPiers** [comment]
Sotto la Pioggia (http://ifitalia.oldgamesitalia.net/pmwiki/pmwiki.php?n=Giochi.SottoLaPioggia): Plays sound

**Comment by PeterPiers** [comment]
Souterraine (http://ifdb.tads.org/viewgame?id=kwda1sxcxxo4twt): Plays sound.

Also, the main title graphic is truncated, but you've already said that's often due to the actual code of the games and you wouldn't be touching the graphics again barring something critical, so that's ok. Just mentioning it for thoroughness.

**Comment by PeterPiers** [comment]
Space InvaderZ (http://ifdb.tads.org/viewgame?id=xmw7lmceqylyt0vp): A ZMachine abuse, so priority is, I would imagine, quite low!

It's far too slow to play, and indeed after opening it doing anything in iFrotz slows to a crawl until a different game is loaded (if it doesn't crash in the meantime). But apart from the speed it seems ok - I wonder whether some optimisations might be possible that'd help with this and other abuses and other more demanding games? Just wondering. Again - super-low priority. Then again, if the optimisations also helped with, say, complex pathfinding in newer games (which always provokes a small delay)...
